### PR TITLE
Handle log and error for unsupported colorspaces

### LINKED ISSUE #1305: Unsupported Colorspace Format Error
This is somewhat related to #1264, but I think it's different enough to warrant its own issue.

I'm using a custom data set, so I can't share it here, but there were 3 images in my data set in the wrong colorspace format, and TC properly warned me about them and told me which images were the offending ones, as expected. After deleting them, a second issue occurs:
```
Traceback (most recent call last):
  File "make_predictions.py", line 14, in <module>
    sf['image'] = tc.image_analysis.resize(sf['image'], 150, 200)
  File "/usr/local/lib/python2.7/site-packages/turicreate/toolkits/image_analysis/image_analysis.py", line 159, in resize
    return image.apply(lambda x: _extensions.resize_image(x, width, height, channels, decode, resample_method))
  File "/usr/local/lib/python2.7/site-packages/turicreate/data_structures/sarray.py", line 1894, in apply
    dryrun = [fn(i) for i in self.head(100) if i is not None]
  File "/usr/local/lib/python2.7/site-packages/turicreate/toolkits/image_analysis/image_analysis.py", line 159, in <lambda>
    return image.apply(lambda x: _extensions.resize_image(x, width, height, channels, decode, resample_method))
  File "/usr/local/lib/python2.7/site-packages/turicreate/extensions.py", line 171, in <lambda>
    return lambda *args, **kwargs: _run_toolkit_function(fn, arguments, args, kwargs)
  File "/usr/local/lib/python2.7/site-packages/turicreate/extensions.py", line 160, in _run_toolkit_function
    raise _ToolkitError(ret[1])
turicreate.toolkits._main.ToolkitError: Unsupported colorspace format. Currently, only RGB and Grayscale are supported.
```

This is fine, and I understand that only RGB and Grayscale formats are supported, but there's no additional information here on which images are the offenders. I was able to solve this by using the `channels` attribute on `tc.Image` with a binary filter on my SFrame:

```python
sf = sf[sf['image'].apply(lambda img: img.channels <= 3)]
```

I still don't think this is the best solution because formats like YUV and HSV will still "pass" this test. Is there any way to get additional debugging information that tells us which images are causing trouble?

### PR DESCRIPTION
**Pull request by fareeha**
This PR fixes the issues emerging from changes made in PR #1212 'Handle resizing of channels of input images'.

Change allows unsupported colorspaces to be detected while loading images. URL for the offending image is now logged with the error message. 

Fixes #1305 and #1279

### CODE REVIEW

File: src/image/jpeg_io.cpp [hunk 1]
```diff
@@ -36,7 +36,7 @@ jpeg_error_exit (j_common_ptr cinfo)
 
 void parse_jpeg(const char* data, size_t length,
                 size_t& width, size_t& height,
-                size_t& channels) {
+                size_t& channels, const std::string& url) {
```
Comment (nickjong) [inline src/image/jpeg_io.cpp]: Where is the `url` argument used?

File: src/image/jpeg_io.cpp [hunk 2]
```diff
@@ -50,6 +50,10 @@ void parse_jpeg(const char* data, size_t length,
     jpeg_mem_src(&cinfo, (unsigned char*)data, length); // Specify data source for decompression
     jpeg_read_header(&cinfo, TRUE); // Read file header, set default decompression parameters
 
+    if (cinfo.out_color_space != JCS_GRAYSCALE && cinfo.out_color_space != JCS_RGB) {
+      log_and_throw(std::string("Unsupported colorspace format. Currently, only RGB and Grayscale are supported."));
```
Comment (nickjong) [inline src/image/jpeg_io.cpp]: How does this interact with the `ignore_failure` parameter of `load_images` (in `src/unity/lib/image_util.cpp`?

Comment (fareeha) [reply src/image/jpeg_io.cpp]: If ignore_failure parameter is set to true then it logs the error above along with the url of the offending file.

image_util.cpp Line [188:](https://github.com/apple/turicreate/blob/af42916ca1b2816843c56b749e4166e6a6d709b6/src/unity/lib/image_util.cpp#L188)  logprogress_stream << error << "\t" << " file: " << sanitize_url(*iter) << std::endl;

which logs the error as:

"Unsupported colorspace format. Currently, only RGB and Grayscale are supported.	 file: ./images/sample.jpg"

Comment (nickjong) [reply src/image/jpeg_io.cpp]: Ah, got it, I see now. Thanks for the explanation!

File: src/image/jpeg_io.cpp (hunk 1 above)
Comment (fareeha) [reply src/image/jpeg_io.cpp]: This is redundant - my bad. Thank you for the catch! I'll fix this.

image_util has access to url of images which it uses to log the error. So we don't need to include url as a parameter in parse_jpeg.

**Pull request merged**